Update OTel Operator readme and Target Allocator readme (#1951)
open-telemetry/opentelemetry-operator: cmd/otel-allocator/README.md

# Target Allocator

Target Allocator is an optional component of the OpenTelemetry Collector [Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) (CR). The release version matches the
operator's most recent release as well.

> In a nutshell, the TA is used to manage Prometheus metrics without actually needing to have Prometheus installed in your Kubernetes cluster.

Reviewer: I don't think this is a correct assessment.  The target allocator does no scraping or management of Prometheus metrics.  It is a mechanism for decoupling the service discovery and metric collection functions of Prometheus such that they can be scaled independently.  The collector is the component that manages Prometheus metrics without needing to install Prometheus.  The TA manages the configuration of the collector's prometheus receiver.

The TA serves two functions:
* Even distribution of Prometheus targets among a pool of Collectors
* Discovery of Prometheus Custom Resources

## Even Distribution of Prometheus Targets

> The Target Allocator's first job is to evenly distribute [Prometheus targets](https://www.tigera.io/learn/guides/prometheus-monitoring/prometheus-metrics/#:~:text=Prometheus%20monitoring%20works%20by%20identifying,and%20generates%20metrics%20from%20it.) among a pool of OTel Collectors. This means that the OTel Collectors collect the metrics instead of a Prometheus [scraper](https://uzxmx.github.io/prometheus-scrape-internals.html). Metrics are ingested by the OTel Collectors by way of the [Prometheus Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/README.md).

Reviewer: The first job is to discover both targets to scrape and collectors to allocate targets to.  Then it can distribute the targets it discovers among the collectors.

https://uzxmx.github.io/prometheus-scrape-internals.html is probably not the best link for this statement as the OTel collector actually uses all of those scraper internals in its prometheus receiver.

## Discovery of Prometheus Custom Resources

> The Target Allocator’s second job is the discovery of [Prometheus CRs](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md), namely the [ServiceMonitor and PodMonitor](https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator#target-allocator). The ServiceMonitor and the PodMonitor don’t do any scraping themselves; their purpose is to inform the Target Allocator (or Prometheus) to add a new job to their scrape configuration. These metrics are then ingested by way of the Prometheus Receiver on the OpenTelemetry Collector.

Reviewer: ```suggestion
The Target Allocator also provides for the discovery of [Prometheus Operator CRs](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md), namely the [ServiceMonitor and PodMonitor](https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator#target-allocator). The ServiceMonitor and the PodMonitor don’t do any scraping themselves; their purpose is to inform the Target Allocator (or Prometheus) to add a new job to their scrape configuration. These metrics are then ingested by way of the Prometheus Receiver on the OpenTelemetry Collector.
```
